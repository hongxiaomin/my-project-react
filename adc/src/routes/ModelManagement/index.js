import ModelRoute from './routes/ModelManagement'

import Model from './components/Model'

export default store => ({
  path: 'model',
  indexRoute: {
    component: Model,
  },
  /*  Async getComponent is only invoked when route matches   */
  getChildRoutes(location, cb) {
    require.ensure([], () => {
      cb(null, [
        ModelRoute(store),
      ])
    })
  },
})
