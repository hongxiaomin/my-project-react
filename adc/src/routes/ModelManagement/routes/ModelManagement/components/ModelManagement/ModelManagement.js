import React from 'react'
import { connect } from 'react-redux'
import Mqtt from 'mqtt'
import { isEmpty } from 'lodash'
import { injectIntl } from 'react-intl'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import AlertMqttError from 'components/AlertMqttError'
import AlertModelMqttError from 'components/AlertModelMqttError'
import ModelList from '../ModelList'
import ModelDetail from '../ModelDetail'
import CreateModel from '../CreateModel'
import Train from '../Train'
import Test from '../Test'
import classes from './ModelManagement.scss'
import {
  TRAINING_CHANGE_MESSAGE,
  TESTING_CHANGE_MESSAGE,
  getModelList,
  getLicenseList,
} from '../../modules'

type Props = {
  pageStatus: String,
  modelDetail: Object,
  dispatch: Function,
  modelList: Array,
  licenseList: Array,
  intl: Object,
};

export class ModelManagement extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      isDetail: false,
      row: {},
      progress: 0,
      mqttError: 'loading',
      visible: false,
      isError: false,
      errorMsg: '',
      modelId: '',
      trainingPercent: '0',
      testingPercent: '0',
    }
    this.client = null
    this.isDetail = this.isDetail.bind(this)
  }

  componentDidMount() {
    this.props.dispatch(getModelList())
    this.props.dispatch(getLicenseList())
    this.subscriptionMqttWithDeviceId()
  }

  componentWillUnmount() {
    // UnSubscribe mqtt
    if (this.client) {
      this.client.end(() => {
        console.log('Disconnect')
      })
    }
  }

  subscriptionMqttWithDeviceId = () => {
    let mqttClient
    if (__PROD__) {
      mqttClient = Mqtt.connect(`wss://${location.hostname}:59001`, {
        rejectUnauthorized: false,
        username: 'deltaww',
        password: 'Dare2Define',
      })
    } else {
      mqttClient = Mqtt.connect(__MQTT__, {
        rejectUnauthorized: false,
        username: 'deltaww',
        password: 'Dare2Define',
      })
    }

    mqttClient.stream.on('error', () => {
      this.setState({
        visible: true,
        mqttError: false,
      })
      mqttClient.end()
    })

    mqttClient.on('connect', () => {
      this.setState({
        mqttError: true,
      })
      mqttClient.subscribe('autojudge/training/info')
      mqttClient.subscribe('autojudge/testing/info')
    })
    mqttClient.on('message', (topic, message) => {
      const msgArray = JSON.parse(message.toString())
      const state = {
        modelId: msgArray.modelId.toString(),
      }
      const obj = {
        payload: msgArray,
      }
      console.log('message', msgArray)
      if (msgArray.status === 'error') {
        this.setState({
          isError: true,
          errorMsg: msgArray.divaErrorMessage,
        })
      }
      if (topic === 'autojudge/training/info') {
        state.trainingPercent = msgArray.percent
        obj.type = TRAINING_CHANGE_MESSAGE
      }
      if (topic === 'autojudge/testing/info') {
        state.testingPercent = msgArray.percent
        obj.type = TESTING_CHANGE_MESSAGE
      }

      this.setState(state)
      this.props.dispatch(obj)
    })
    this.client = mqttClient
  }

  isDetail(isOpen, rowDetail) {
    this.setState({
      isDetail: isOpen,
      row: rowDetail._original,
    })
  }

  handleMqttError = () => {
    this.setState({
      visible: false,
    })
  }
  handleModelMqttError = () => {
    this.setState({
      isError: false,
    })
  }
  props: Props
  render() {
    const { intl } = this.props
    const number = this.props.modelList.filter((item) => {
      return (item.currentState !== 'none')
    })
    let cards = []
    if (!isEmpty(this.props.licenseList)) {
      cards = this.props.licenseList.filter(item => item.isTrainingDevice === true && item.isActivate === true)
    }
    const gpuArr = cards.map(item => {
      return item.deviceGpus.map(i => {
        return i.trainingServiceCount
      })
    })
    const gpuNumber = gpuArr.toString().split(',').reduce((ret, item) => Number(ret) + Number(item), 0)
    return (
      <div className={classes.container}>
        <div className={classes.top}>
          <h4 className={classes.model}>
            <WrapperFormattedMessage
              id="nav.modelManagement"
            />
          </h4>
          <CreateModel />
        </div>
        {
          cards.length === 0 ?
            <h5 style={{ margin: '15px 0px' }}><WrapperFormattedMessage id='modelManagement.remainingService' />: 0</h5> :
            <h5 style={{ margin: '15px 0px' }}><WrapperFormattedMessage id='modelManagement.remainingService' />: {gpuNumber - number.length}</h5>
        }
        <ModelList isDetail={this.isDetail} />
        {
          !isEmpty(this.props.modelDetail) &&
          this.state.isDetail &&
          this.props.pageStatus === 'detailPage' &&
          <ModelDetail
            rowDetail={this.state.row}
            isDetail={this.isDetail}
            modelDetail={this.props.modelDetail}
            trainingPercent={this.state.trainingPercent}
            testingPercent={this.state.testingPercent}
            modelId={this.state.modelId}
          />
        }
        {
          this.props.pageStatus === 'startTrainPage' &&
          <Train rowDetail={this.state.row} />
        }
        {
          this.props.pageStatus === 'startTestPage' &&
          <Test rowDetail={this.state.row} />
        }
        <AlertMqttError isOpen={this.state.visible} isError={this.state.mqttError} handleCancel={this.handleMqttError} />
        <AlertModelMqttError isOpen={this.state.isError} handleCancel={this.handleModelMqttError} errorMsg={this.state.errorMsg} />
      </div>
    )
  }
}
export default injectIntl(connect(
  state => ({
    token: state.auth.token,
    pageStatus: state.modelManagement.pageStatus,
    modelList: state.modelManagement.modelList,
    modelDetail: state.modelManagement.modelDetail,
    licenseList: state.modelManagement.licenseList,
  }),
)(ModelManagement))
