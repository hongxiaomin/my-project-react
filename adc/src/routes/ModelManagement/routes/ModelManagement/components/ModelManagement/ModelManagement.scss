.container {
  padding: 1% 1%;
  margin: 20px 20px ;
  // display: flex;
}
.top {
  display: flex;
  justify-content: flex-start;
}
.model {
  margin-right: 10px;
  padding-top: 3px;
}
.searchDiv {
 display: flex;
 justify-content: flex-start;
 margin-top: 20px;
}
.singleSearch {
  display: flex;
  justify-content: flex-start;
  margin-right: 5% ;
}
.h5 {
  padding-top: 6px;
   margin-right: 10px;
}
.select {
  cursor: pointer;
  outline: 0;
  border: 0;
  background: inherit;
  box-shadow: none;
  margin: 0;
  border: 1px solid #D9D9D9;
  height: 30px;
  width: 200px;
  border-radius: 3px;
}
