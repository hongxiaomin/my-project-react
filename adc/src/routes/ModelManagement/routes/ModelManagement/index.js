import { injectReducer } from 'store/reducers'

export default store => ({
  path: 'modelManagement',
  getComponent(nextState, cb) {
    /*  Webpack - use 'require.ensure' to create a split point
        and embed an async module loader (jsonp) when bundling   */
    require.ensure([], (require) => {
      /*  Webpack - use require callback to define
          dependencies for bundling   */
      const ModelManagement = require('./components/ModelManagement').default
      const { modelManagement } = require('./modules/index')
      /*  Add the reducer to the store on key 'counter'  */
      injectReducer(store, { key: 'modelManagement', reducer: modelManagement })

      /*  Return getComponent   */
      cb(null, ModelManagement)

    /* Webpack named bundle   */
    }, 'modelManagement')
  },
})
