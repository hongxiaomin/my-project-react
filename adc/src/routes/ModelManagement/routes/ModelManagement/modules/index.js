/* eslint no-param-reassign: 0 */
import { showMessage, stringToFirstUpperCase } from 'Utils'
import { assign, map, isEmpty, uniqWith, isEqual } from 'lodash'
import { simpleFetch } from 'helpers/fetchWrapper'

// const __HOST__ = 'http://172.22.40.161:5005/'
// const __HOST__ = 'http://172.22.34.10:5005/'
// ------------------------------------
// Constants
// ------------------------------------
export const MODEL_LIST = 'MODEL_LIST'
export const MODEL_DETAIL = 'MODEL_DETAIL'
export const UPDATE_MODELLIST_TRAIN = 'UPDATE_MODELLIST_TRAIN'
export const UPDATE_MODELLIST_TEST = 'UPDATE_MODELLIST_TEST'
export const TRAINING_DETAIL = 'TRAINING_DETAIL'
export const TRAINING_CHANGE_MESSAGE = 'TRAINING_CHANGE_MESSAGE' // 推送来的消息
export const TESTING_CHANGE_MESSAGE = 'TESTING_CHANGE_MESSAGE'
export const TESTING_DETAIL = 'TESTING_DETAIL'
export const PAGE_STATUS = 'PAGE_STATUS' // 跳转切换页面
export const IMAGE_LIST = 'LABELINGTOOL.IMAGE_LIST'
export const SEARCH_IMAGE_LIST = 'SEARCH_IMAGE_LIST'
export const PRODUCT_PROCESS_CODE = 'PRODUCT_PROCESS_CODE'
export const GET_MODEL_TEST_RESULT_TYPE = 'GET_MODEL_TEST_RESULT_TYPE'
export const UPDATE_MODEL_TEST_RESULT_TYPE = 'UPDATE_MODEL_TEST_RESULT_TYPE'
export const UPDATE_ALL_MODEL_TEST_RESULT_TYPE = 'UPDATE_ALL_MODEL_TEST_RESULT_TYPE'
export const TESTING_TOTAL_COUNT = 'TESTING_TOTAL_COUNT'
export const IMPORT_MODEL_FLAG = 'IMPORT_MODEL_FLAG'
export const ALL_DEFCTCODE = 'ALL_DEFCTCODE'
export const TEST_RESULT_LIST = 'TEST_RESULT_LIST'
export const LOAD_FILE_MESSAGE = 'LOAD_FILE_MESSAGE'
export const GET_LICENSE_LIST = 'GET_LICENSE_LIST'
export const COPY_MODEL_DETAIL = 'COPY_MODEL_DETAIL'
export const VIEW_ID = 'VIEW_ID'
// ------------------------------------
// Async actions
// ------------------------------------
const imageList = [
  {
    imageId: 1,
    name: 'image1',
    path: '/test1.jpg',
    isLabeled: true,
  },
  {
    imageId: 2,
    name: 'image2',
    path: '/test2.jpg',
    isLabeled: true,
  },
  {
    imageId: 3,
    name: 'image3',
    path: '/test1.jpg',
    isLabeled: false,
  },
]


export const updateCopyModelDetail = obj => (dispatch, getState) => {
  dispatch({
    type: COPY_MODEL_DETAIL,
    data: {
      ...getState().modelManagement.copyModelDetail,
      ...obj,
    },
  })
}

export const imageListCheck = (params, type) => (dispatch, getState) => {
  return simpleFetch('labeling/label/search', {
    method: 'POST',
    body: params,
  })
  .then((json) => {
    if (json.code === 200000) {
      if (type === 'edit') {
        console.log('getState', getState().modelManagement.copyModelDetail)
        const detail = getState().modelManagement.copyModelDetail

        const targetCode = JSON.parse(json.result.targetCode)
        const othersCode = JSON.parse(json.result.othersCode)
        dispatch({
          type: COPY_MODEL_DETAIL,
          data: {
            ...detail,
            imageSourceCondition: {
              ...detail.imageSourceCondition,
              targetCode,
              othersCode,
            },
          },
        })
      } else {
        dispatch({
          type: ALL_DEFCTCODE,
          data: {
            targetCode: JSON.parse(json.result.targetCode),
            othersCode: JSON.parse(json.result.othersCode),
          },
        })
      }
    }
  })
}

export const getImageByPath = paths => (dispatch) => {
  // 通过path获取图片路径， path为array
  return simpleFetch('model/images', {
    method: 'POST',
    body: paths,
  })
    .then((data) => {
      if (data.code === 200000) {
        showMessage({ text: stringToFirstUpperCase(data.message), type: 'success' })
        dispatch({
          type: TESTING_TOTAL_COUNT,
          data: data.result.total,
        })
      } else {
        showMessage({ text: stringToFirstUpperCase(data.message), type: 'error' })
      }
    })
}

// 获取model列表
export const getModelList = () => (dispatch) => {
    // 请求获取model
  return simpleFetch('models')
  .then((data) => {
    if (data.code === 200000) {
      dispatch({
        type: MODEL_LIST,
        payload: data.result,
      })
    }
  })
}

// 新建model
export const createModel = (state, cb, cb2) => (dispatch) => {
  return simpleFetch('model', {
    method: 'POST',
    body: state,
  })
  .then((json) => {
    if (json.code === 200000) {
      if (cb) {
        cb()
      }
      dispatch(getModelList())
      showMessage({ text: json.message, type: 'success' })
    } else {
      if (cb2) {
        cb2()
      }
      showMessage({ text: json.message, type: 'error' })
    }
  })
}

// copy model
export const copyModel = (state, cb, cb2) => (dispatch, getState) => {
  const { intl: { messages } } = getState()
  return simpleFetch('model', {
    method: 'POST',
    body: state,
  })
  .then((json) => {
    if (json.code === 200000) {
      if (cb) {
        cb()
      }
      dispatch(getModelList())
      showMessage({ text: messages['modelManguage.copy.success'], type: 'success' })
    } else {
      if (cb2) {
        cb2()
      }
      showMessage({ text: json.message, type: 'error' })
    }
  })
}
// 编辑model
export const updateModel = (data, cb, cb2) => (dispatch) => {
  return simpleFetch('model', {
    method: 'PUT',
    body: data,
  })
  .then((json) => {
    if (json.code === 200000) {
      if (cb) {
        cb()
      }
      dispatch(getModelList())
      showMessage({ text: json.message, type: 'success' })
    } else {
      if (cb2) {
        cb2()
      }
      showMessage({ text: json.message, type: 'error' })
    }
  })
}

// 导入model
export const importModel = state => (dispatch) => {
  return simpleFetch('model/default', {
    method: 'POST',
    body: state,
  })
  .then((json) => {
    if (json.code === 200000) {
      dispatch(getModelList())
      showMessage({ text: json.message, type: 'success' })
    } else {
      showMessage({ text: json.message, type: 'error' })
    }
  })
}

// 发布下架model
export const actionModel = state => (dispatch) => {
  return simpleFetch('model', {
    method: 'PUT',
    body: state,
  })
  .then((json) => {
    if (json.code === 200000) {
      showMessage({ text: json.message, type: 'success' })
      dispatch(getModelList())
    } else {
      showMessage({ text: json.message, type: 'error' })
    }
  })
}
// 设置默认model
export const defaultModel = state => (dispatch) => {
  return simpleFetch('model/default', {
    method: 'PUT',
    body: state,
  })
  .then((json) => {
    if (json.code === 200000) {
      showMessage({ text: json.message, type: 'success' })
      dispatch(getModelList())
    } else {
      showMessage({ text: json.message, type: 'error' })
    }
  })
}
// 删除 model
export const deleteModel = state => (dispatch) => {
  return simpleFetch('model', {
    method: 'DELETE',
    body: { modelId: state },
  })
  .then((json) => {
    if (json.code === 200000) {
      showMessage({ text: json.message, type: 'success' })
      dispatch(getModelList())
    } else {
      showMessage({ text: json.message, type: 'error' })
    }
  })
}
export const getLicenseList = () => dispatch =>
  simpleFetch('devices')
  .then((res) => {
    if (res.code === 200000) {
      dispatch({
        type: GET_LICENSE_LIST,
        payload: JSON.parse(res.result),
      })
    }
  })

// 训练或者测试中的model的操作
export const actionTrainModel = (state, cb) => (dispatch) => {
  if (state.command === 'start') {
    return simpleFetch('model/progress', {
      method: 'POST',
      body: state,
    })
    .then((json) => {
      if (json.code === 200000) {
        showMessage({ text: json.message, type: 'success' })
        if (cb) {
          cb()
        }
        const tmp = {
          id: state.modelId,
          deviceHash: json.result.deviceHash,
          viewId: json.result.viewId,
        }
        dispatch({
          type: VIEW_ID,
          data: tmp,
        })
      } else {
        if (cb) {
          cb()
        }
        showMessage({ text: json.message, type: 'error' })
      }
    })
  } else if (state.command === 'restart') {
    return simpleFetch('model/progress', {
      method: 'POST',
      body: state,
    })
    .then((json) => {
      if (json.code === 200000) {
        showMessage({ text: json.message, type: 'success' })
        if (cb) {
          cb()
        }
      } else {
        if (cb) {
          cb()
        }
        showMessage({ text: json.message, type: 'error' })
      }
    })
  } else {
    console.log('state', state)
    return simpleFetch('model/progress', {
      method: 'POST',
      body: state,
    })
    .then((json) => {
      if (json.code === 200000) {
        if (cb) {
          cb()
        }
        dispatch(getLicenseList())
        showMessage({ text: json.message, type: 'success' })
      } else {
        if (cb) {
          cb()
        }
        showMessage({ text: json.message, type: 'error' })
      }
    })
  }
}


// 获取每行model的detail
export const getModelDetail = state => (dispatch) => {
  return simpleFetch(`model?modelId=${state}`)
  .then((json) => {
    if (json.code === 200000) {
      dispatch({
        type: MODEL_DETAIL,
        payload: json.result,
      })
    }
  })
}

// 获取training时的detail信息
export const getModelTrainDetail = id => (dispatch) => {
  return simpleFetch(`model/progress?modelId=${id}&state=training`)
  .then((data) => {
    if (data.code === 200000) {
      dispatch({
        type: UPDATE_MODELLIST_TRAIN,
        payload: Object.assign(data.result, { id }),
      })
    }
  })
}

// 获取testing时的detail信息
export const getModelTestDetail = id => (dispatch) => {
    // 请求获取model
  return simpleFetch(`model/progress?modelId=${id}&state=testing`)
  .then((data) => {
    if (data.code === 200000) {
      dispatch({
        type: UPDATE_MODELLIST_TEST,
        payload: Object.assign(data.result, { id }),
      })
    }
  })
}

export const getSelectInfo = () => (dispatch) => {
  // 请求获取product, process, code 信息
  return simpleFetch('labeling/image/search')
    .then((data) => {
      if (data.code !== 200000) {
        showMessage({ text: data.message, type: 'error' })
        return
      }
      const result = data.result
      dispatch({
        type: PRODUCT_PROCESS_CODE,
        data: result,
      })
    })
    .catch((e) => {
      console.log('e:', e)
    })
}

export const getTestResultList = id => (dispatch) => {
  return simpleFetch(`model/testing/result?modelId=${id}`)
  .then((data) => {
    if (data.code === 200000) {
      dispatch({
        type: TEST_RESULT_LIST,
        data: data.result,
      })
    }
  })
  .catch(err => console.log(err))
}

export function getTestResultType(id) {
  return (dispatch) => {
    // simpleFetch('/api/defectCode')
    return simpleFetch(`model/defect?modelId=${id}`)
    // .then(response => response.json())
    .then((data) => {
      if (data.code === 200000) {
        dispatch({
          type: GET_MODEL_TEST_RESULT_TYPE,
          data: data.result,
        })
      }
    })
    .catch(err => console.log(err))
  }
}

export function updateConfidence(state) {
  return () => {
    simpleFetch('model/defect', {
      method: 'PUT',
      body: state,
    })
    .then((json) => {
      if (json.code === 200000) {
        showMessage({ text: json.message, type: 'success' })
      } else {
        showMessage({ text: json.message, type: 'error' })
      }
    })
    .catch(err => console.log(err))
  }
}

const ACTION_HANDLERS = {
  [MODEL_LIST]: (state, action) => {
    const tmp = action.payload
    tmp.map((item) => {
      if (!isEmpty(item.targetCodeId)) {
        item.targetCodeId = JSON.parse(item.targetCodeId)
        item.targetCodeId = item.targetCodeId.map(i => {
          return i.toString()
        })
      }
      if (!isEmpty(item.othersCodeId)) {
        item.othersCodeId = JSON.parse(item.othersCodeId)
        item.othersCodeId = item.othersCodeId.map(i => {
          return i.toString()
        })
      }
      if (!isEmpty(item.imageSourceCondition) && item.sourceType === 'filter') {
        item.imageSourceCondition = JSON.parse(item.imageSourceCondition)
      }
      return item
    })
    return assign({}, state, {
      modelList: tmp,
    })
  },
  [VIEW_ID]: (state, action) => {
    const modelList = state.modelList.map((item) => {
      let tmp = Object.assign({}, item)
      if (tmp.id == action.data.id) {
        tmp = { ...tmp,
          deviceHash: action.data.deviceHash,
          viewId: action.data.viewId,
        }
      }
      return tmp
    })
    return assign({}, state, { modelList })
  },
  [TEST_RESULT_LIST]: (state, action) => {
    return assign({}, state, { testResultList: action.data })
  },
  [COPY_MODEL_DETAIL]: (state, action) => {
    return assign({}, state, { copyModelDetail: action.data })
  },
  [MODEL_DETAIL]: (state, action) => assign({}, state, {
    modelDetail: action.payload,
  }),
  [TESTING_TOTAL_COUNT]: (state, action) => assign({}, state, {
    testingImageCount: action.data,
  }),
  [GET_LICENSE_LIST]: (state, action) => assign({}, state, {
    licenseList: action.payload,
  }),
  // 点击train详情以后接口返回的数据，把数据放进modellist里面。（方便后续推送操作以及页面的更新）
  [UPDATE_MODELLIST_TRAIN]: (state, action) => {
    const trainingDetail = action.payload
    const modelList = state.modelList.map((item) => {
      let tmp = Object.assign({}, item)
      if (tmp.id === action.payload.id) {
        tmp = Object.assign({}, item, { trainingDetail })
      }
      return tmp
    })
    return assign({}, state, { modelList }, { trainingDetail })
  },
  // 点击test详情以后接口返回的数据，把数据放进modellist里面。（方便后续推送操作以及页面的更新）
  [UPDATE_MODELLIST_TEST]: (state, action) => {
    const testingDetail = action.payload
    const modelList = state.modelList.map((item) => {
      let tmp = Object.assign({}, item)
      if (tmp.id == action.payload.id) {
        tmp = Object.assign({}, item, { testingDetail })
      }
      return tmp
    })
    return assign({}, state, { modelList }, { testingDetail })
  },
  // 获取到后台推送来的消息，显示到页面并且刷新training的status，以及modellist的status
  [TRAINING_CHANGE_MESSAGE]: (state, action) => {
    const trainingDetail = action.payload
    const modelList = state.modelList.map((item) => {
      let tmp = Object.assign({}, item)
      if (tmp.id == action.payload.modelId) {
        tmp = Object.assign({}, item, { trainingDetail })
        tmp = { ...tmp,
          status: action.payload.status,
          currentState: action.payload.currentState,
          isTrained: action.payload.isTrained,
          isVerified: action.payload.isVerified,
          divaErrorMessage: action.payload.divaErrorMessage,
        }
      }
      return tmp
    })
    return assign({}, state, { modelList })
  },
  // 获取到后台推送来的消息，显示到页面并且刷新testing的status，以及modellist的status
  [TESTING_CHANGE_MESSAGE]: (state, action) => {
    const testingDetail = action.payload
    const modelList = state.modelList.map((item) => {
      let tmp = Object.assign({}, item)
      if (tmp.id == action.payload.modelId) {
        tmp = Object.assign({}, item, { testingDetail })
        tmp = { ...tmp,
          status: action.payload.status,
          currentState: action.payload.currentState,
          isTrained: action.payload.isTrained,
          isVerified: action.payload.isVerified,
        }
      }
      return tmp
    })
    return assign({}, state, { modelList })
  },
  // 当前处在哪个页面。
  [PAGE_STATUS]: (state, action) => {
    return assign({}, state, {
      pageStatus: action.payload,
    })
  },
  [ALL_DEFCTCODE]: (state, action) => {
    return assign({}, state, {
      allDefectCode: action.data,
    })
  },
  [PRODUCT_PROCESS_CODE]: (state, action) => {
    return Object.assign(
      {},
      state,
      {
        defectCodeList: action.data.defect,
        productList: action.data.product,
        processList: action.data.process,
      },
    )
  },
  [IMAGE_LIST]: (state, action) => {
    return Object.assign({}, state, { imageList: action.data })
  },
  [SEARCH_IMAGE_LIST]: (state, action) => {
    return Object.assign({}, state, { searchImageList: action.data })
  },
  [GET_MODEL_TEST_RESULT_TYPE]: (state, action) => {
    return assign({}, state, {
      modelResultType: action.data,
    })
  },
  [UPDATE_MODEL_TEST_RESULT_TYPE]: (state, action) => {
    let { modelResultType } = state
    const { confidence, keyIndex } = action.data
    modelResultType = map(modelResultType, (element, index) => {
      return (keyIndex === index) ? assign({}, element, { defaultConfidence: confidence }) : element
    })
    return assign({}, state, { modelResultType })
  },
}

const initialState = {
  modelList: [],
  modelDetail: {},
  pageStatus: 'detailPage',
  imageList: [],
  searchImageList: [],
  defectCodeList: [],
  productList: [],
  processList: [],
  testingImageCount: 0,
  allDefectCode: {},
  testResultList: [],
  licenseList: [],
  copyModelDetail: {
    description: '',
    imageSourceCondition: {
      dateRange: {
        startTime: '',
        endTime: '',
      },
      othersCode: [],
      processId: '',
      productId: [],
      targetCode: [],
      targetCodeId: [],
      othersCodeId: [],
    },
    iteration: 100,
    lossRate: 0.00001,
    name: '',
    sourceType: 'path',
  },
}

export const modelManagement = (state = initialState, action) => {
  const handler = ACTION_HANDLERS[action.type]

  return handler ? handler(state, action) : state
}
