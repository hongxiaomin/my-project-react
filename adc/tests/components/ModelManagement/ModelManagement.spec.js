import React from 'react'
import { shallow } from 'enzyme'
import { ModelManagement } from 'routes/ModelManagement/routes/ModelManagement/components/ModelManagement/ModelManagement'

describe('(Component) ModelManagement', () => {
  it('render normally', () => {
    const modelList = [{ id: 1, name: 'a' }]
    const licenseList = [{ id: 1, deviceGpus: { serviceCount: 11 } }]
    const wrapper = shallow(
      <ModelManagement modelList={modelList} licenseList={licenseList} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render modelDetail is null', () => {
    const modelDetail = {}
    const modelList = [{ id: 1, name: 'a' }]
    const licenseList = [{ id: 1, deviceGpus: { serviceCount: 11 } }]
    const wrapper = shallow(
      <ModelManagement modelDetail={modelDetail} modelList={modelList} licenseList={licenseList} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render modelDetail is not null', () => {
    const modelDetail = {
      trainingInfo: { trainingEndTime: '', trainingBeginTime: '', trainingProgress: 0 },
    }
    const modelList = [{ id: 1, name: 'a' }]
    const licenseList = [{ id: 1, deviceGpus: { serviceCount: 11 } }]
    const wrapper = shallow(
      <ModelManagement modelDetail={modelDetail} modelList={modelList} licenseList={licenseList} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render pageStatus === startTrainPage', () => {
    const pageStatus = 'startTrainPage'
    const modelList = [{ id: 1, name: 'a' }]
    const licenseList = [{ id: 1, deviceGpus: { serviceCount: 11 } }]
    const wrapper = shallow(
      <ModelManagement pageStatus={pageStatus} modelList={modelList} licenseList={licenseList} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  
})
