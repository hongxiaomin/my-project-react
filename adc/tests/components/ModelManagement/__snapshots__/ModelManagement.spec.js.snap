// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ModelManagement render modelDetail is not null 1`] = `
<div
  className="container"
>
  <div
    className="top"
  >
    <h4
      className="model"
    >
      Model Management
    </h4>
    <CreateModel />
  </div>
  <h5
    style={
      Object {
        "margin": "15px 0px",
      }
    }
  >
    Remaining Service: 0
  </h5>
  <Connect(ModelList)
    isDetail={[Function]}
  />
  <AlertMqttError
    handleCancel={[Function]}
    isError="loading"
    isOpen={false}
  />
  <AlertModelMqttError
    errorMsg=""
    handleCancel={[Function]}
    isOpen={false}
  />
</div>
`;

exports[`(Component) ModelManagement render modelDetail is null 1`] = `
<div
  className="container"
>
  <div
    className="top"
  >
    <h4
      className="model"
    >
      Model Management
    </h4>
    <CreateModel />
  </div>
  <h5
    style={
      Object {
        "margin": "15px 0px",
      }
    }
  >
    Remaining Service: 0
  </h5>
  <Connect(ModelList)
    isDetail={[Function]}
  />
  <AlertMqttError
    handleCancel={[Function]}
    isError="loading"
    isOpen={false}
  />
  <AlertModelMqttError
    errorMsg=""
    handleCancel={[Function]}
    isOpen={false}
  />
</div>
`;

exports[`(Component) ModelManagement render normally 1`] = `
<div
  className="container"
>
  <div
    className="top"
  >
    <h4
      className="model"
    >
      Model Management
    </h4>
    <CreateModel />
  </div>
  <h5
    style={
      Object {
        "margin": "15px 0px",
      }
    }
  >
    Remaining Service: 0
  </h5>
  <Connect(ModelList)
    isDetail={[Function]}
  />
  <AlertMqttError
    handleCancel={[Function]}
    isError="loading"
    isOpen={false}
  />
  <AlertModelMqttError
    errorMsg=""
    handleCancel={[Function]}
    isOpen={false}
  />
</div>
`;

exports[`(Component) ModelManagement render pageStatus === startTrainPage 1`] = `
<div
  className="container"
>
  <div
    className="top"
  >
    <h4
      className="model"
    >
      Model Management
    </h4>
    <CreateModel />
  </div>
  <h5
    style={
      Object {
        "margin": "15px 0px",
      }
    }
  >
    Remaining Service: 0
  </h5>
  <Connect(ModelList)
    isDetail={[Function]}
  />
  <Connect(Train)
    rowDetail={Object {}}
  />
  <AlertMqttError
    handleCancel={[Function]}
    isError="loading"
    isOpen={false}
  />
  <AlertModelMqttError
    errorMsg=""
    handleCancel={[Function]}
    isOpen={false}
  />
</div>
`;
